public/systems_monitoring_graphs.js

```diff
@@ -0,0 +1,641 @@
+                    )
+            }
+
+            // What to do when one group is hovered
+            let highlight = function(d){
+                console.log(d)
+                // reduce opacity of all groups
+                d3.selectAll(".diskArea").style("opacity", .1)
+                // expect the one that is hovered
+                d3.select("."+d).style("opacity", 1)
+            }
+
+            // And when it is not hovered anymore
+            let noHighlight = function(d){
+                d3.selectAll(".diskArea").style("opacity", 1)
+            }
+
+            // Add legend for each name.
+            let size = 20
+            svg.selectAll("myrect")
+                .data(diskKeys)
+                .enter()
+                .append("rect")
+                    .attr("x", Width + 10)
+                    .attr("y", (d,i) => 10 + i*(size+5)) // 10 is where the first dot appears. 25 is the distance between dots
+                    .attr("width", size)
+                    .attr("height", size)
+                    .style("fill", d => color(d))
+                    .on("mouseover", highlight)
+                    .on("mouseleave", noHighlight)
+
+            // Add legend labels for each name.
+            svg.selectAll("mylabels")
+                .data(diskMetrics)
+                .enter()
+                .append("text")
+                    .attr("x", Width + 10 + size*1.2)
+                    .attr("y", (d,i) => 10 + i*(size+5) + (size/2)) // 10 is where the first dot appears. 25 is the distance between dots
+                    .style("fill", d => color(d))
+                    .text(d => d)
+                    .attr("text-anchor", "left")
+                    .style("alignment-baseline", "middle")
+                    .on("mouseover", highlight)
+                    .on("mouseleave", noHighlight)
+
+            // Disk usage graph title
+            svg.append("text")
+                .attr("x", Width / 2)
+                .attr("y", 0 - Margin.top / 2)
+                .attr("text-anchor", "middle")
+                .style("font-size", "20px")
+                .style("text-decoration", "bold")
+                .text("Disk Usage");
+        }
+
+        function plotMemoryMetrics(data) {
+            // Append svg to element with class chart2
+            let svg = d3.select(".chart2")
+                .append("svg")
+                .attr("width", Width + Margin.left + Margin.right + 120)
+                .attr("height", Height + Margin.top + Margin.bottom)
+                .append("g")
+                .attr("transform",
+                    "translate(" + Margin.left + "," + Margin.top + ")");
+
+            // Add X axis
+            let x = d3.scaleTime().domain(d3.extent(data, d => new Date(d.datetime))).range([1, Width]);
+            let xAxis = svg.append("g")
+                .attr("transform", "translate(0," + Height + ")")
+                .call(d3.axisBottom(x).tickFormat(dayTimeFormat).ticks(13))
```

Comment:
This still doesn't seem to be okay. The graph should be per day as the data streams in e.g 10 mins of traffic graphs.

Reply:
Updated x axis time scale to be in per day